### PR TITLE
Handle change in blockdev size on setup

### PR DESCRIPTION
**Pull request by mulkieran**
Resolves: #764
Resolves: #435 

Supercedes: #787

### CODE REVIEW

File: src/engine/strat_engine/backstore/blockdevmgr.rs [hunk 1]
```diff
@@ -451,8 +449,8 @@ fn initialize(pool_uuid: PoolUuid,
             // The dev_size is at least MIN_DEV_SIZE, but the size of the
             // metadata is not really bounded from above.
             let blockdev =
-                StratBlockDev::new(dev, devnode.to_owned(), dev_size, bda, &[], None, hw_id)
-                    .expect("bda.size() < dev_size and no user facing segments allocated");
+                StratBlockDev::new(dev, devnode.to_owned(), bda, &[], None, hw_id)
+                    .expect("bda.size() == dev_size; only allocating space for metadata");
```
Comment (mulkieran) [inline src/engine/strat_engine/backstore/blockdevmgr.rs]: I updated the expect to be more accurate.

File: src/engine/strat_engine/backstore/blockdevmgr.rs [hunk 2]
```diff
@@ -433,13 +433,11 @@ fn initialize(pool_uuid: PoolUuid,
...
     let mut bds: Vec<StratBlockDev> = Vec::new();
     for (dev, (devnode, dev_size, mut f)) in add_devs {
-        let dev_size = dev_size.sectors();
-
         let bda = BDA::initialize(&mut f,
                                   pool_uuid,
                                   Uuid::new_v4(),
                                   mda_size,
-                                  dev_size,
+                                  dev_size.sectors(),
```
Comment (mulkieran) [inline src/engine/strat_engine/backstore/blockdevmgr.rs]: I got rid of the let above, since the value is only used once.

**Comment by mulkieran** [comment]
@agrover review? Changes between this and prelim version have been commented.

**Comment by agrover** [comment]
:+1:

**Pull request merged**